### ISSUE TITLE
Numpy like Masked/where operations

### ISSUE THREAD

**Issue by bipin-lekhak** [pytorch/pytorch#116461]
### 🚀 The feature, motivation and pitch

Numpy allows me to do masked operations such as mean/variance using `where` argument for mask:

```python
mean_value = np.mean(x, where=mask)
var_value = np.mean(x, where=mask) 
```

[Numpy docs for same](https://numpy.org/doc/stable/reference/generated/numpy.ma.mean.html)

This format of defining mean and variance is much more concise and less error inducing than the one decribed for pythorch [here](https://stackoverflow.com/a/69314298/4488940).

Having something similar for pytorch would be extremely valuable in making these tasks less error-prone and concise.

**Comment by vadimkantorov** [comment]
Related:
- https://github.com/pytorch/pytorch/issues/69359

**Comment by malfet** [comment]
@vadimkantorov should this issue be closed as duplicate of https://github.com/pytorch/pytorch/issues/69359?

**Comment by vadimkantorov** [comment]
maybe? but I guess the best would be also placing somewhere in README/docs the status of `torch._masked` or of `nan*` ops. There is https://pytorch.org/maskedtensor/main/index.html, but I think it's outdated

**Comment by vadimkantorov** [comment]
@malfet btw https://github.com/pytorch/maskedtensor / torch._masked seems deprecated/archived, but https://pytorch.org/maskedtensor/main/index.html is still published as if it was maintained

**Comment by vadimkantorov** [comment]
To address the OP, this seems to work:
```
>>> torch.masked.mean(torch.tensor([1,2,3]), mask = torch.tensor([True, False, True]))
tensor(2.)
```

**Comment by bipin-lekhak** [comment]
Thanks. I tried doing something like:

```python
loss = F.mse_loss(y, yhat)
loss_masked = torch.masked.masked_tensor(loss, mask)
loss_mean = loss_masked.mean()
```

But I had a hard time trying to properly backprop this way.

I have not checked with the above code, but I think my earlier implementation might be relevant to this discussion.

**Comment by vadimkantorov** [comment]
also, maybe you would need to pass `F.mse_loss(..., reduction = 'none')`...
